### PR TITLE
Add fix function for #919

### PR DESCRIPTION
**Pull request by joaquimg**
fix candidate for #919

### CODE REVIEW

File: src/JuMP.jl [hunk 1]
```diff
@@ -391,6 +391,14 @@ function setvalue(v::Variable, val::Number)
     end
 end
 
+# Fix a variable that was not previously fixed
+function fix(v::Variable, val::Number)
+    v.m.colVal[v.col] = val
```
Comment (mlubin) [inline src/JuMP.jl]: Could ``setvalue`` and ``setcategory`` instead of the internal methods?

**Comment by mlubin** [comment]
Needs tests and docs.

**Comment by joaquimg** [comment]
This one runs smoothly:
```function fix(v::Variable, val::Number)
    v.m.colCat[v.col] = :Fixed
    setvalue(v, val)
end```
While this one does not work:
```function fix(v::Variable, val::Number)
    setcategory(v, :Fixed)
    setvalue(v, val)
end```
Due to:
```
const var_cats = [:Cont, :Int, :Bin, :SemiCont, :SemiInt]
function setcategory(v::Variable, cat::Symbol)
    cat in var_cats || error("Unrecognized variable category $cat. Should be one of:\n    $var_cats")
    v.m.colCat[v.col] = cat
end
```
Clearly ``:Fixed`` is not allowed...

**Comment by mlubin** [comment]
Okay, you could argue that using ``setcategory`` is not appropriate for fixed variables because you should also update the bounds at the same time to maintain consistency. I wonder if we should remove the special behavior from ``setvalue`` and just use ``fix`` to update the values of fixed variables.

**Comment by joaquimg** [comment]
I think the current behavior is ambiguous and makes a little confusion, although I added the docs.

I would prefer ``setvalue`` only for warmstarts and ``fix`` for fixed values (allowing the user to fix previously non-fixed variables)

**Comment by joaquimg** [comment]
Lets go for  

> remove the special behavior from `setvalue` and just use `fix` to update the values of fixed variables.

?

**Comment by mlubin** [comment]
Yes. Calling ``setvalue`` on a fixed variable should (for now) throw an error since this is a breaking change and we don't want code silently changing behavior.

File: src/JuMP.jl [hunk 2]
```diff
@@ -386,15 +386,16 @@ getupperbound(v::Variable) = v.m.colUpper[v.col]
 function setvalue(v::Variable, val::Number)
     v.m.colVal[v.col] = val
     if v.m.colCat[v.col] == :Fixed
-        v.m.colLower[v.col] = val
-        v.m.colUpper[v.col] = val
+        error("setvalue use for setting bounds of fixed variables was deprecated in favour of JuMP.fix(v::Variable, val::Number), which changes bounds of fixed variables and fixes previously not fixed variables ")
```
Comment (mlubin) [inline src/JuMP.jl]: Deprecated would mean that we still support it but throw a warning. This isn't the case. ``s/favour/favor/``

Comment (joaquimg) [inline src/JuMP.jl]: what do I say?
`setvalue use for setting bounds of fixed variables was made obsolete in favor of JuMP.fix(v::Variable, val::Number), which changes bounds of fixed variables and fixes previously not fixed variables`

Comment (mlubin) [inline src/JuMP.jl]: ``setvalue for fixed variables is no longer supported. Use JuMP.fix instead.``

File: doc/refvariable.rst [hunk 3]
```diff
@@ -219,7 +219,9 @@ Two possible uses for fixed variables are:
    the sensitivity of the objective with respect to the fixed value may be queried with ``getdual(x)``.
 
 2. For solving a sequence of problems with varying parameters.
-   One may call ``setvalue(x, val)``
+   One may call ``JuMP.fix(x, val)``
    to change the value to which the variable is fixed. For LPs
    in particular, most solvers are able to efficiently hot-start when
    solving the resulting modified problem.
+   
+Note that ``setvalue(x, val)`` will not set initial value for fixed variables, it will only set a warmstart for non-fixed variables. One can fix existing, non-fixed variables with ``JuMP.fix(x, val)`` which sets both bounds and initial value of ``x`` to ``val``.
```
Comment (mlubin) [inline doc/refvariable.rst]: I don't think this paragraph is needed

File: doc/refvariable.rst [hunk 4]
```diff
@@ -219,7 +219,9 @@ Two possible uses for fixed variables are:
    the sensitivity of the objective with respect to the fixed value may be queried with ``getdual(x)``.
 
 2. For solving a sequence of problems with varying parameters.
-   One may call ``setvalue(x, val)``
+   One may call ``JuMP.fix(x, val)``
```
Comment (mlubin) [inline doc/refvariable.rst]: One may call ``JuMP.fix(x,val)`` to change the value the which the variable is fixed or to fix a previously unfixed variable.

**Comment by mlubin** [comment]
Almost ready to merge, but it's currently failing tests:
```
Test Summary:                                 | Pass  Error  Total
  Variables                                   |   85      1     86
  constructors                                |   11            11
  get and set bounds                          |   11      1     12
```

**Comment by mlubin** [comment]
There are still failing tests

**Comment by joaquimg** [comment]
I will look into that!

**Comment by joaquimg** [comment]
I had no errors in Variables, also I could not notice errors in Xpress and GLPK

**Comment by mlubin** [comment]
See https://travis-ci.org/JuliaOpt/JuMP.jl/jobs/190050309#L409

**Comment by joaquimg** [comment]
Now its passing all tests!

**Pull request merged**